pkg/api/dashboard.go

```diff
@@ -195,9 +196,15 @@ func deleteDashboard(c *models.ReqContext) Response {
 	}
 
 	err := dashboards.NewService().DeleteDashboard(dash.Id, c.OrgId)
-	if err == models.ErrDashboardCannotDeleteProvisionedDashboard {
-		return Error(400, "Dashboard cannot be deleted because it was provisioned", err)
-	} else if err != nil {
+	if err != nil {
+		var dashboardErr models.DashboardErr
+		if ok := errors.As(err, &dashboardErr); ok {
+			// XXX: Should we allow other dashboard errors also?
```

Comment:
@papagian I added that comment. Do you mind if we leave it? I added it since I'm unsure if it actually makes sense to 500 on all dashboard errors except for `ErrDashboardCannotDeleteProvisionedDashboard`.

Reply:
ok then

Reply:
Actually, on second look, let's remove the comment. I see the connection now.